src/neo/Ledger/Blockchain.cs

```diff
@@ -87,16 +87,23 @@ static Blockchain()
         {
             GenesisBlock.RebuildMerkleRoot();
 
-            NativeContract[] contracts = { NativeContract.GAS, NativeContract.NEO };
             using (ScriptBuilder sb = new ScriptBuilder())
             {
-                foreach (NativeContract contract in contracts)
+                foreach (NativeContract contract in new NativeContract[] { NativeContract.GAS, NativeContract.NEO })
                 {
                     sb.EmitAppCall(contract.Hash, "onPersist");
                     sb.Emit(OpCode.DROP);
                 }
-
-                onPersistNativeContractScript = sb.ToArray();
+                onPersistScript = sb.ToArray();
+            }
+            using (ScriptBuilder sb = new ScriptBuilder())
+            {
+                foreach (NativeContract contract in new NativeContract[] { NativeContract.Oracle })
+                {
+                    sb.EmitAppCall(contract.Hash, "postPersist");
```

Comment:
Do you think that it's feasiable move `onPersist` of GAS and NEO to `postPersist`? we can burn the fee of the transaction before the execution of this TX and save one foreach. The difference it's that the reward could be used in the next block.

Reply:
`postPersist` will be executed after **all** transactions have been executed.

Reply:
Maybe we can store in `onPersist` the amount that the oracle will mint in the next block and reward them in the next block, also, we can remove the response data  when response it's executed. 

I'm trying to avoid 3 transaction `foreach`, it could reduce the tps.

Reply:
> we can remove the response data when response it's executed.

The response callback may fail so that the response won't be removed.

Reply:
We can send two snapshots to the execution, one with the clone, and other with the persisted data, and in this syscall we can use a different snapshot, what do you think?, also we can clean the data in the next good execution.

Reply:
Why should we do this? I think `postPersist()` is good.